Content.Shared/_Moffstation/Extensions/EntitySystemExt.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Runtime.CompilerServices;
+using Robust.Shared.Utility;
+
+namespace Content.Shared._Moffstation.Extensions;
+
+public static class EntitySystemExt
+{
+    /// Throws a debug assert and logs the given <paramref name="message"/> to the system's error logger.
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static void AssertOrLogError(this EntitySystem entSys, string message)
+    {
+        DebugTools.Assert(message);
+        entSys.Log.Error(message);
+    }
+
+    /// <see cref="AssertOrLogError"/>, but returns <paramref name="ret"/> instead of void.
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static T AssertOrLogError<T>(this EntitySystem entSys, string message, T ret)
+    {
+        DebugTools.Assert(message);
+        entSys.Log.Error(message);
+        return ret;
+    }
+}
```

Comment:
I made these to more easily do the "log an error or explode if we're in debug" which I was putting all over the place.